### PR TITLE
Add direct links to for-statement section of C# language spec

### PR DESCRIPTION
**Pull request by gewarren**
Added three direct links to the for-statement section of the C# language spec. Also added more section headings as I found it a bit difficult to parse the topic to find the info on the parts of a for loop.

### CODE REVIEW

File: docs/csharp/language-reference/keywords/for.md [hunk 1]
```diff
@@ -32,16 +34,18 @@ By using a `for` loop, you can run a statement or a block of statements repeated
...
+Note that, if the initial value of `i` is greater than 5, the body of the loop doesn't run even once.
+
+## Sections of a for statement
   
- Every `for` statement defines initializer, condition, and iterator sections. These sections usually determine how many times the loop iterates.  
+Every [for statement](/dotnet/csharp/language-reference/language-specification/statements#the-for-statement) defines *initializer*, *condition*, and *iterator* sections. These sections usually determine how many times the loop iterates.  
   
 ```csharp  
 for (initializer; condition; iterator)  
     body
```
Comment (mairaw) [inline docs/csharp/language-reference/keywords/for.md]: not sure if it's this line that is causing the rest of the formatting here on GH to look funny but everything else seems to be interpreted as code

Comment (BillWagner) [reply docs/csharp/language-reference/keywords/for.md]: It looks correct to me (in GH and on the review site)

Comment (gewarren) [reply docs/csharp/language-reference/keywords/for.md]: @mairaw I don't think I see what you're seeing. [It looks okay](https://review.docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/for?branch=pr-en-us-5405) to me?

image

**Review by BillWagner — COMMENTED** [review]
This LGTM.

I'll let @mairaw :shipit: based on her comments in the one section.

**Review by BillWagner — APPROVED** [review]

File: docs/csharp/language-reference/keywords/for.md (hunk 1 above)
Comment (mairaw) [reply docs/csharp/language-reference/keywords/for.md]: It’s probably an issue with the GitHub diff functionality.

1605d2af-1c72-4e74-be11-4eef3b5544f0

**Pull request merged**